annotated.go

```diff
@@ -687,12 +761,19 @@ func (la *lifecycleHookAnnotation) buildHookInstaller(ann *annotated) (
 	invokeParamTypes := []reflect.Type{
 		_typeOfLifecycle,
 	}
+	if err := la.validateHookDeps(origHookFnT, paramTypes, resultTypes); err != nil {
```

Comment:
I notice in `validateHookDeps` we are checking for and skipping context.Context. The for loop directly after this call is already going through the params of the hook fn and removing context from them. Should we call `validateHookDeps` after the loop below and pass `invokeParamTypes` in instead of `origHookFnT` to not duplicate this work?

Inserting this call here also confuses the comment above about removing context, since that's not the purpose of this call.